annotations/component-annotations/src/main/java/io/ap4k/component/generator/CompositeConfigGenerator.java

```diff
@@ -16,35 +16,39 @@
  **/
 package io.ap4k.component.generator;
 
+import io.ap4k.Generator;
+import io.ap4k.WithProject;
 import io.ap4k.component.adapter.CompositeConfigAdapter;
 import io.ap4k.component.annotation.CompositeApplication;
 import io.ap4k.component.config.CompositeConfig;
 import io.ap4k.component.configurator.ApplyProject;
 import io.ap4k.component.handler.ComponentHandler;
 import io.ap4k.component.handler.ComponentServiceCatalogHandler;
-import io.ap4k.Generator;
-import io.ap4k.WithProject;
 import io.ap4k.config.ConfigurationSupplier;
 
 import javax.lang.model.element.Element;
 import java.util.Map;
 
 public interface CompositeConfigGenerator extends Generator, WithProject {
-
+  String GENERATOR_KEY = "composite";
+  
   @Override
   default void add(Map map) {
-    on(new ConfigurationSupplier<>(CompositeConfigAdapter.newBuilder(propertiesMap(map, CompositeApplication.class)).accept(new ApplyProject(getProject()))));
+    add(new ConfigurationSupplier<>(
+      CompositeConfigAdapter
+        .newBuilder(propertiesMap(map, CompositeApplication.class))
+        .accept(new ApplyProject(getProject()))));
   }
 
   @Override
   default void add(Element element) {
     CompositeApplication composite = element.getAnnotation(CompositeApplication.class);
-    on(composite != null
+    add(composite != null
        ? new ConfigurationSupplier<>(CompositeConfigAdapter.newBuilder(composite).accept(new ApplyProject(getProject())))
       : new ConfigurationSupplier<>(CompositeConfig.newCompositeConfigBuilder().accept(new ApplyProject(getProject()))));
   }
 
-  default void on(ConfigurationSupplier<CompositeConfig> config) {
```

Comment:
The convention used across generators is that the `add` methods get the initial input and create a supplier out of it and the `on` method just processes the supplier. 

It doesn't really matter as its something strictly internal, I am just mentioning it.

Reply:
@iocanel I actually changed it because it's called `add` in `KubernetesApplicationGenerator`, https://github.com/ap4k/ap4k/blob/master/annotations/kubernetes-annotations/src/main/java/io/ap4k/kubernetes/generator/KubernetesApplicationGenerator.java#L71